src/sound.cpp

```diff
@@ -30,9 +32,10 @@ static void OpenBankFile(const char *filename)
 	/* If there is no sound file (nosound set), don't load anything */
 	if (filename == nullptr) return;
 
-	FioOpenFile(SOUND_SLOT, filename, BASESET_DIR);
-	size_t pos = FioGetPos();
-	uint count = FioReadDword();
+	delete _original_sound_file;
+	_original_sound_file = new RandomAccessFile(filename, BASESET_DIR);
```

Comment:
Does this still need to be global? It doesn't appear to be used outside the OpenBankFile function, and at the bottom of this function it's then saved within the SoundEntry object but any reuse of this variable causes that data to be deleted anyway? Seems wrong

Reply:
The first thing that happens in OpenBankFile is clear all the SoundEntry objects that it was used in (memset). So, there is no SoundEntry that refers to it anymore once we arrive at `delete` in OpenBankFile.